### PR TITLE
Feature/automated testing

### LINKED ISSUE #3: Test services with travis
Services already support basic testing, just need to build a simple unit tests for them

### LINKED ISSUE #4: Write docs!
We could use some basic documentation with reference to simple_http as an example

### PR DESCRIPTION
**Pull request by omercnet**
Patching up services
Introducing automated testing (only plugins for now)

fixes #3 
fixes #4

### CODE REVIEW

File: services/banner/banner_service.py [hunk 1]
```diff
@@ -1,52 +1,85 @@
...
 
-    def _send_alert(self, data, originating_ip, originating_port, dest_port):
+    def _send_alert(self, originating_ip, originating_port):
         params = {
-            defs.AlertFields.EVENT_TYPE.name: 'banner_port_access',
-            defs.AlertFields.ORIGINATING_IP.name: originating_ip,
-            defs.AlertFields.ORIGINATING_PORT.name: originating_port,
-            defs.AlertFields.DEST_PORT.name: dest_port,
-            defs.AlertFields.EVENT_DESC.name: data,
+            'event_type': BANNER_ALERT_TYPE_NAME,
```
Comment (Sleeptime17) [inline services/banner/banner_service.py]: " "

File: services/banner/banner_service.py [hunk 2]
```diff
@@ -1,52 +1,85 @@
...
+            'originating_ip': originating_ip,
```
Comment (Sleeptime17) [inline services/banner/banner_service.py]: " "

File: services/banner/banner_service.py [hunk 3]
```diff
@@ -1,52 +1,85 @@
...
-        port = int(self.service_args["listening_port"])
-        banner = self.service_args["banner"].decode('string_escape')
-        self.server = banner_server.MultiSocketServer([port], banner, self._send_alert)
+        """Start banner service."""
+        requestHandler = BannerRequestHandler
+        requestHandler.banner = self.service_args.get(BANNER)
+        requestHandler.alert = self._send_alert
+
+        port = int(self.service_args.get(PORT))
+        self.server = ThreadingTCPServer(('', port), requestHandler)
```
Comment (Sleeptime17) [inline services/banner/banner_service.py]: " "

File: services/banner/banner_service.py [hunk 4]
```diff
@@ -1,52 +1,85 @@
...
+            self.server = None
 
+    def test(self):
+        """Test service alerts and return a list of triggered event types."""
+        event_types = list()
 
+        self.logger.debug('executing service test')
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.settimeout(3)
+        s.connect(('127.0.0.1', int(self.service_args.get(PORT))))
```
Comment (Sleeptime17) [inline services/banner/banner_service.py]: " "

File: services/intel_amt/intel_amt_service.py [hunk 5]
```diff
@@ -124,7 +129,20 @@ def on_server_start(self):
         self.signal_ready()
         self.server.serve_forever()
 
+    def test(self):
+        """trigger service alerts and return a list of triggered event types"""
+
+        event_types = [BUSYBOX_TELNET_AUTHENTICATION]
+        url = 'http://127.0.0.1:{}/index.htm'.format(AMT_PORT)
```
Comment (Sleeptime17) [inline services/intel_amt/intel_amt_service.py]: I think I did this, but " "

File: services/intel_amt/intel_amt_service.py [hunk 6]
```diff
@@ -124,7 +129,20 @@ def on_server_start(self):
...
+        requests.get(url, headers={'Authorization': 'username="test"'})
+        event_types.append(AMT_AUTH_ATTEMPT_ALERT_TYPE)
+        requests.get(url, headers={'Authorization': 'response=""'})
```
Comment (Sleeptime17) [inline services/intel_amt/intel_amt_service.py]: Also "" just for the 'Authorization' here, and 2 lines above this

File: services/asa/README.md [hunk 7]
```diff
@@ -1,43 +0,0 @@
-# Cicso ASA honeypot
```
[the remark this answers is not in the document]
Comment (omercnet) [reply services/asa/README.md]: it's not fully compatible yet, we need to make it (specifically the IKE library) work with Python2

File: services/banner/banner_service.py (hunk 1 above)
Comment (omercnet) [reply services/banner/banner_service.py]: no, we're not doing that anymore.

File: services/intel_amt/intel_amt_service.py [hunk 8]
```diff
@@ -124,7 +129,20 @@ def on_server_start(self):
         self.signal_ready()
         self.server.serve_forever()
 
+    def test(self):
+        """trigger service alerts and return a list of triggered event types"""
+
+        event_types = [BUSYBOX_TELNET_AUTHENTICATION]
```
Comment (omercnet) [inline services/intel_amt/intel_amt_service.py]: this looks unrelated @Sleeptime17

**Comment by omercnet** [comment]
fucking a
@Cymmetria/honeycomb please review

File: README.rst [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    :alt: Updates
+.. |Codecov| image:: https://img.shields.io/codecov/c/github/Cymmetria/honeycomb_plugins.svg
+   :alt: Codecov
+   :target: https://codecov.io/gh/Cymmetria/honeycomb_plugins
+
+=================
+Honeycomb Plugins
+=================
+
+This is the plugin repository for Honeycomb_, the honeypot framework by Cymmetria_.
```
[the remark this answers is not in the document]
Comment (omercnet) [reply README.rst]: it's a rst thing, it's a reference link, see the few lines below
just helps with keeping things clean and also lets you use that reference in other places 
http://docutils.sourceforge.net/docs/user/rst/quickref.html#hyperlink-targets
